### PR TITLE
Reset and minimize button

### PR DESCRIPTION
**Pull request by ptomato**
https://phabricator.endlessm.com/T23957

### CODE REVIEW

File: src/framework/toolbox.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ const {Toolbox} = imports.toolbox;
 
 var FrameworkToolbox = GObject.registerClass(class FrameworkToolbox extends Toolbox {
     _init(props = {}) {
+        props.title = 'Hack Modules';
```
Comment (cosimoc) [inline src/framework/toolbox.js]: Should be marked for translation.

File: src/toolbox.js [hunk 2]
```diff
@@ -4,6 +4,11 @@ const {GObject, Gtk} = imports.gi;
 
 var Toolbox = GObject.registerClass({
     CssName: 'toolbox',
+    Properties: {
+        title: GObject.ParamSpec.string('title', 'Title', '',
+            GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT,
+            'Hack'),
```
Comment (cosimoc) [inline src/toolbox.js]: Should also be marked for translation.

File: src/toolbox.js [hunk 3]
```diff
@@ -12,14 +17,59 @@ var Toolbox = GObject.registerClass({
...
-        this.add(headerbar);
+        this._headerbar = new Gtk.HeaderBar({
+            hasSubtitle: false,
+            title: this._title,
+        });
+        const image = new Gtk.Image({iconName: 'view-refresh-symbolic'});
+        const buttonReset = new Gtk.Button({
+            alwaysShowImage: true,
+            image,
+            label: 'Reset All',
```
Comment (cosimoc) [inline src/toolbox.js]: Should also be marked for translation

File: src/toolbox.js [hunk 4]
```diff
@@ -12,14 +17,59 @@ var Toolbox = GObject.registerClass({
...
-        const buttonReset = new Gtk.Button();
-        const image = new Gtk.Image({stock: 'gtk-revert-to-saved'});
-        buttonReset.add(image);
-        headerbar.pack_end(buttonReset);
-        headerbar.show_all();
-        this.add(headerbar);
+        this._headerbar = new Gtk.HeaderBar({
+            hasSubtitle: false,
+            title: this._title,
+        });
```
Comment (cosimoc) [inline src/toolbox.js]: Could be a bit cleaner to create this before chaining up, so you can remove the `if (this._headerbar)` check in the setter below?

**Review by cosimoc — CHANGES_REQUESTED** [review]

**Comment by ptomato** [comment]
Fixed.

**Pull request merged**